src/main/scala/com/microsoft/hyperspace/index/IndexManager.scala

```diff
@@ -96,4 +96,12 @@ trait IndexManager {
    * @return all indexes that match any of the given states
    */
   def getIndexes(states: Seq[String] = Seq()): Seq[IndexLogEntry]
+
+  /**
+   * Get index metadata and detailed index statistics for a given index.
+   *
+   * @param indexName Name of the index to get stats for.
+   * @return Index metadata and statistics as a [[DataFrame]].
+   */
+  def getIndexStats(indexName: String): DataFrame
```

Comment:
How about adding a utility function - `printIndexStats` which prints the result dataframe in a transposed way?  e.g.
```
name | index1
indexedColumns | [RGUID]
includedColumns | [Date]
numBuckets | 200
schema | "type":"struct","fields":[{"name":"RGUID","type":"string","nullable":true,"metadata":{}}, {"name":"Date","type":"string","nullable":true,"metadata":{}},{"name":"_data_file_id","type":"long","nullable":false,"metadata":{}}]}|
...
```
It's because - the result dataframe has only 1 rows and it can be a little bit inconvenient to check stats in a line.

Reply:
I think such beautification is really up to the consumer. I am not sure adding more and more APIs and user facing functions into the codebase is a good idea. DataFrame is a standard output and easy to use. User can easily interpret/convert it into any desired format/representation.

Reply:
But if they are using Notebook or spark-shell, they need to write such function every time. I think 1-2 utility functions would be good to have. In any case it's just for usability and let's add the print function later if needed.